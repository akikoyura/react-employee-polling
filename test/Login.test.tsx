import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import {store} from "../src/store/store";
import {fireEvent, render} from "@testing-library/react";
import {getAppData} from "../src/actions/actions";
import Login from "../src/components/Login";

describe("Login", () => {
    it("should be rendered successfully", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Login/>
                </BrowserRouter>
            </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    })

    it ('should successfully log in after the submit button is clicked', async () => {
        await store.dispatch(getAppData());

        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Login/>
                </BrowserRouter>
            </Provider>
        );

        expect(component).toBeDefined();

        const loginHeadingElement = component.getByTestId("login-heading");
        const usernameInputElement = component.getByTestId("username");
        const passwordInputElement = component.getByTestId("password");
        const submitButtonElement = component.getByTestId("submit");

        expect(loginHeadingElement).toBeInTheDocument();
        expect(usernameInputElement).toBeInTheDocument();
        expect(passwordInputElement).toBeInTheDocument();
        expect(submitButtonElement).toBeInTheDocument();

        fireEvent.change(usernameInputElement, {target: {value: 'johndoe'}});
        fireEvent.change(passwordInputElement, {target: {value: '123456'}});
        expect(usernameInputElement.value).toBe("johndoe");
        expect(passwordInputElement.value).toBe("123456");
        fireEvent.click(submitButtonElement);

    })
})