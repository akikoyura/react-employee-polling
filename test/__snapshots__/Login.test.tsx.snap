// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login > should be rendered successfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="d-flex justify-content-center mb-3 mt-5 col-sm-12"
          >
            <h3>
              Employee Polls
            </h3>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="d-flex justify-content-center col-sm-12"
          >
            <div
              class="card"
              style="width: 540px;"
            >
              <div
                class="card-body"
              >
                <div
                  class="d-flex justify-content-center card-title h5"
                  data-testid="login-heading"
                >
                  Login
                </div>
                <form
                  class=""
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="username"
                    >
                      User name: 
                    </label>
                    <input
                      class="form-control"
                      data-testid="username"
                      id="username"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="password"
                    >
                      Password: 
                    </label>
                    <input
                      class="form-control"
                      data-testid="password"
                      id="password"
                      type="password"
                      value=""
                    />
                  </div>
                  <button
                    class="btn btn-primary d-block w-100 btn btn-primary"
                    data-testid="submit"
                    type="submit"
                  >
                    Login
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="d-flex justify-content-center mb-3 mt-5 col-sm-12"
        >
          <h3>
            Employee Polls
          </h3>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="d-flex justify-content-center col-sm-12"
        >
          <div
            class="card"
            style="width: 540px;"
          >
            <div
              class="card-body"
            >
              <div
                class="d-flex justify-content-center card-title h5"
                data-testid="login-heading"
              >
                Login
              </div>
              <form
                class=""
              >
                <div
                  class="mb-3"
                >
                  <label
                    class="form-label"
                    for="username"
                  >
                    User name: 
                  </label>
                  <input
                    class="form-control"
                    data-testid="username"
                    id="username"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="mb-3"
                >
                  <label
                    class="form-label"
                    for="password"
                  >
                    Password: 
                  </label>
                  <input
                    class="form-control"
                    data-testid="password"
                    id="password"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  class="btn btn-primary d-block w-100 btn btn-primary"
                  data-testid="submit"
                  type="submit"
                >
                  Login
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
